test/services/MessageSummaryServiceSpec.scala

```diff
@@ -271,6 +273,60 @@ class MessageSummaryServiceSpec extends SpecBase with ModelGenerators with Scala
     }
   }
 
+  "cancellationDecisionMessage" - {
+
+    "must return" - {
+      "None when there are none in the movement" in {
+        forAll(ie009Gen) {
+          ie009 =>
+            forAll(createMovement(NonEmptyList.one(ie009))) {
+              departure =>
+                service.cancellationDecisionMessage(departure) must not be defined
+
+            }
+        }
+      }
+
+      "latest IE009 when there is only an IE015 and a IE009" in {
+        forAll(ie015Gen, ie009Gen) {
+          (ie015, ie009) =>
+            forAll(createMovement(NonEmptyList.of(ie015, ie009))) {
+              departure =>
+                val (message, messageId) = service.cancellationDecisionMessage(departure).value
+
+                message mustEqual ie009
+                messageId mustEqual MessageId.fromMessageIdValue(2).value
+            }
+        }
+      }
+
+      "None when there has been a cancellation decision message and correction" in {
```

Comment:
Based on the above comment this test will fail. It'll need to be updated to pass back a cancellation decision